### PR TITLE
Add TLS key/certificate support

### PR DESCRIPTION
**Pull request by sulkaharo**
Adds support for TLS key/certificate authentication, as used by the Finnish Kanta services (see https://www.kanta.fi/en/system-developers/my-kanta-pages-phr).

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -16,10 +16,12 @@ class Client {
    * @param {String} config.baseUrl ISS for FHIR server
    * @param {Object} config.customHeaders Optional custom headers to send with
    *   each request
+   * @param {String} config.key TLS key for FHIR server
+   * @param {String} config.cert TLS certificate for FHIR server
```
Comment (bkaney) [inline lib/client.js]: These are optional so we should have `[]`.

Comment (sulkaharo) [reply lib/client.js]: Fixed :)

Comment (bkaney) [reply lib/client.js]: Or I guess say “Optional” like the param.

**Comment by sulkaharo** [comment]
Unit test would we awesome - I tried looking for samples for how to set it up but I didn't come up with an easy way to mock the certificate validation. OTOH the feature relies on just passing the parameters to request-native which actually implements the feature, so I'm not sure what the success/fail criteria for such tests would be?

**Comment by bkaney** [comment]
I think a test that asserts setting the key and cert in the constructor sets the appropriate params when setting the bearer token.

**Comment by bkaney** [comment]
Hi @sulkaharo -- Where are the specs on this (e.g. passing around a key and cert in each request)?

**Comment by bkaney** [comment]
I am thinking about this more, perhaps we have an optional parameter that is an object `requestOptions` (instead of hardcoding `key` and `cert`):

```javascript
constructor({ baseUrl, customHeaders = {}, requestOptions = {} })
```

**Comment by bkaney** [comment]
yes, this looks good. need to figure out that failing test.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Node FHIR client library
 * Modern ES6 Classes
 * TDD with Mocha
 * URL polyfill (so it works in client-only apps without much trouble)
+* Support TLS key/certificate pair for REST authentication
```
Comment (bkaney) [inline README.md]: Maybe "Support optional parameters for the request, such as TLS key and cert"

**Pull request merged**